terraform/ec2/linux/main.tf

```
@@ -24,6 +24,9 @@ resource "aws_key_pair" "aws_ssh_key" {
 locals {
   ssh_key_name        = var.ssh_key_name != "" ? var.ssh_key_name : aws_key_pair.aws_ssh_key[0].key_name
   private_key_content = var.ssh_key_name != "" ? var.ssh_key_value : tls_private_key.ssh_key[0].private_key_pem
+  // Canary downloads latest binary. Integration test downloads binary connect to git hash.
+  downloadBinary = var.is_canary ? "aws s3 cp s3://${var.s3_bucket}/release/amazon_linux/${var.arc}/latest/${var.binary_name} ."
+    : "aws s3 cp s3://${var.s3_bucket}/integration-test/binary/${var.cwa_github_sha}/linux/${var.arc}/${var.binary_name} ."
```

super minor nit: You could also just store the S3 URI and then in `inline` have it do the s3 cp. Like `aws s3 cp ${local.binary_uri} .`

```suggestion
  binary_uri = var.is_canary ? "s3://${var.s3_bucket}/release/amazon_linux/${var.arc}/latest/${var.binary_name}"
    : "s3://${var.s3_bucket}/integration-test/binary/${var.cwa_github_sha}/linux/${var.arc}/${var.binary_name}"
```